site/content/docs/developing/package-addons.en.md

```diff
@@ -0,0 +1,155 @@
+# Package Addons <span class="version" > added in [v1.21.0](../../blogs/release-v121.en.md) </span>
+
+Copilot supports uploading local files referenced from your addon templates to S3, and replacing the relevant resource properties with the uploaded S3 location.
+On [`copilot svc deploy`](../commands/svc-deploy.en.md) or [`copilot svc package --upload-assets`](../commands/svc-package.en.md), certain fields on supported resources will be updated with an S3 locaton.
+To see the full list of resources that are supported, take a look at the [AWS CLI documentation](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudformation/package.html).
+
+This feature can be used to deploy local Lambda Functions stored in the same repo as another Copilot service.
+For example, to deploy a javascript Lambda Function alongside a copilot service, you can add this resource to your [addon template](./additional-aws-resources.en.md):
+
+???+ note "Example Lambda Function"
+    === "copilot/service-name/addons/lambda.yml"
+
+        ```yaml hl_lines="4"
+          ExampleFunction:
+            Type: AWS::Lambda::Function
+            Properties:
+              Code: lambdas/example/
+              Handler: "index.handler"
+              Timeout: 900
+              MemorySize: 512
+              Role: !GetAtt "ExampleFunctionRole.Arn"
+              Runtime: nodejs16.x
+          ExampleFunctionRole:
```

Comment:
nit: should we remove the role here, just to highlight more the relevant properties `Handler` and `Code`?

Reply:
probably a good idea in this section - i'll leave them in the full example below, though.